src/app/pages/components/calendar/calendar.component.ts

```diff
@@ -0,0 +1,145 @@
+import {Component, ElementRef, Input, OnInit, ViewEncapsulation} from '@angular/core';
+import * as $ from 'jquery';
+import 'fullcalendar';
+import {Options} from 'fullcalendar';
+import * as moment from 'moment';
+
+@Component({
+    template: '<div></div>',
+    selector: 'ngx-calendar',
+    styleUrls: ['./calendar.component.css'],
+    encapsulation: ViewEncapsulation.None,
+})
+export class CalendarComponent implements OnInit {
+
+    @Input() options: Options;
+    text: string;
+    calendarInitiated: boolean;
+    calendarOptions: Object;
+    todayDate: moment.Moment;
+    private YM: string;
+    private YESTERDAY: string;
+    private TODAY: string;
+    private TOMORROW: string;
+
+    constructor(private element: ElementRef) {
+
+    }
+
+    ngOnInit(): void {
+        this.todayDate = moment().startOf('day');
+        this.YM = this.todayDate.format('YYYY-MM');
+        this.YESTERDAY = this.todayDate.subtract(1, 'day').format('YYYY-MM-DD');
```

Comment:
The date format could be extracted to a constant of some sort, as it is reused multiple times